Documentation/ceph-object-bucket-notifications.md

```diff
@@ -0,0 +1,169 @@
+metadata:
+  name: my-topic [1]
+  namespace: my-app-space [2]
+spec:
+   # use one of the following endpoints
+  endpoint: http://my-notification-endpoint:8080 [3]
+  # endpoint: http://my-notification-endpoint:8080/my-topic
+  # endpoint: https://my-notification-endpoint:8443
+  # endpoint: amqp://my-rabbitmq-service:5672/vhost1
+  # endpoint: amqps://my-rabbitmq-service:5671/vhost1
+  # endpoint: kafka://my-kafka-service:9092
+  objectStoreName: my-store [4]
+  objectStoreNamespace: rook-ceph [5]
+  opaqueData: my@email.com [6]
+  persistent: false [7]
+  http: [8]
+    verifySSL: false [9]
+  # amqp: [10]
+  #   verifySSL: false [11]
+  #   ackLevel: broker [12]
+  #   exchange: my-exchange [13]
+  # kafka: [14]
+  #   verifySSL: false [15]
+  #   ackLevel: broker [16]
+  #   useSSL: false [17]
+
+```
+1. `name` of the `CephBucketTopic`
+   - In case of AMQP endpoint, the name is used for the AMQP topic (“routing key” for a topic exchange)
+   - In case of Kafka endpoint, the name is used as the Kafka topic
+1. `namespace`(optional) of the `CephBucketTopic`. Should match the namespace of the CephBucketNotification associated with this CephBucketTopic, and the OBC with the label referencing the CephBucketNotification
+1. `endpoint` to which to send the notifications to. The identity of the endpoint in formatted as a URI with the following schema options: `http[s]`, `amqp[s]`, `kafka`. The schema of the endpoint must match the type of the endpoint spec provided (see below)
+1. `objectStoreName` is the name of the object store in which the topic should be created. This must be the same object store used for the buckets associated with the notifications referencing this topic.
+1. `objectStoreNamespace` is the namespace of the object store in which the topic should be created
+1. `opaqueData` (optional) is added to all notifications triggered by a notifications associated with the topic
+1. `persistent` (optional) indicates whether notifications to this endpoint are persistent (=asynchronous) or sent synchronously (“false” by default)
+1. `http` (optional) hold the spec for an HTTP endpoint. The format of the URI would be: `http[s]://<fqdn>[:<port>][/<resource>]`
+   - port defaults to: 80/443 for HTTP/S accordingly
+1. `verifySSL` indicates whether the RGW is going to verify the SSL certificate of the HTTP server in case HTTPS is used ("true" by default)
+1. `amqp` (required if `endpioint` schema is `amqp` or `amqps`) hold the spec for an AMQP endpoint. The format of the URI would be: `amqp[s]://[<user>:<password>@]<fqdn>[:<port>][/<vhost>]`
+   - port defaults to: 5672/5671 for AMQP/S accordingly
+   - user/password defaults to: guest/guest
+   - user/password may only be provided if HTTPS is used with the RGW. If not, topic creation request will be rejected
+   - vhost defaults to: “/”
+1. `verifySSL` (optional) indicates whether the RGW is going to verify the SSL certificate of the AMQP server in case AMQPS is used ("true" by default)
```

Comment:
I was looking at the design doc and noticed this. It is generally recommended that the default values for Kubernetes CRDs be the default values for the Go lang as well so that we needn't implement special default behavior. Instead, it would be recommended that we change this to be the inverse, i.e., `disableVerifySSL` where the default is false.

Reply:
And after looking a little more closely, I don't see anywhere in the code where we actually set the default to `true`. Let's change this to `disableVerifySSL` instead as mentioned here: https://github.com/rook/rook/pull/8426/files#r740571718. For all protocol types where `verifySSL` is supposed to be default `true`.